kernel/Makefile

```diff
@@ -29,6 +29,7 @@ lib-y += $(strip \
 	errno.o \
 	work_q.o \
 	system_work_q.o \
+	pthread.o \
```

Comment:
can we make this optional, i.e. make it depend on the some Kconfig so we do not build it by default?